src/auth.js

```diff
@@ -0,0 +1,135 @@
+const validationMessage = {
+  requiredEmail: "이메일을 입력해주세요",
+  invalidEmail: "잘못된 이메일 형식입니다.",
+  requiredPassword: "비밀번호를 입력해주세요.",
+  weekPassword: "비밀번호를 8자 이상 입력해주세요.",
+  mismatchedPassword: "비밀번호가 일치하지 않습니다.",
+  requiredNickname: "닉네임을 입력해주세요.",
+};
+
+const isValidEmail = (email) => {
+  const regex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+  return regex.test(email);
+};
+
+const showMessage = (message, messageElement, inputElement) => {
+  messageElement.textContent = message;
+  messageElement.style.display = "inline-block";
+  inputElement.parentNode.classList.add("validation-focus");
+  toggleButtonActive();
+};
+
+const hideMessage = (messageElement, inputElement) => {
+  messageElement.style.display = "none";
+  inputElement.parentNode.classList.remove("validation-focus");
+  toggleButtonActive();
+};
+
+const isFormComplete = () => {
+  const existValiation = document.querySelector(".validation-focus");
+  const fields = [...document.querySelectorAll("input")];
+  const allCompleted = fields.every((field) => field.value.trim() !== "");
+
+  return allCompleted && !existValiation;
+};
+
+const toggleButtonActive = () => {
+  const $submitButton = document.querySelector(".auth-form > button");
+
+  if (isFormComplete()) {
+    $submitButton.classList.add("active");
+    $submitButton.disabled = false;
+    return;
+  }
+
+  $submitButton.classList.remove("active");
+  $submitButton.disabled = true;
+};
+
+export const togglePasswordVisibility = (event) => {
+  const $passwordInput = event.target.parentNode.querySelector("input");
+
+  if (event.target.classList.contains("show")) {
+    event.target.classList.remove("show");
+    $passwordInput.type = "password";
+    return;
+  }
+
+  event.target.classList.add("show");
+  $passwordInput.type = "text";
+};
+
+export const validateEmail = (inputElement, messageElement) => {
+  const email = inputElement.value.trim();
+
+  if (email === "") {
+    showMessage(validationMessage.requiredEmail, messageElement, inputElement);
+    return;
+  }
+
+  if (!isValidEmail(email)) {
+    showMessage(validationMessage.invalidEmail, messageElement, inputElement);
+    return;
+  }
+
+  hideMessage(messageElement, inputElement);
+};
+
+export const validatePassword = (inputElement, messageElement) => {
+  const password = inputElement.value;
+
+  if (password === "") {
+    showMessage(
+      validationMessage.requiredPassword,
+      messageElement,
+      inputElement
+    );
+    return;
```

Comment:
P2:
개인적으로는`toggleButtonAction`함수를 `showMessage`함수에서 호출하지 않고, 아래처럼 하는것이 더 명확할 것 같습니다.

```suggestion
    showMessage(
      validationMessage.requiredPassword,
      messageElement,
      inputElement
    );
    toggleButtonActive()
    return;
```

Reply:
아래처럼도 작성 가능합니다.

```javascript
const VALIDATION_MESSAGE = {
  REQURIED_EMAIL: "이메일을 입력해주세요",
  INVALID_EMAIL: "잘못된 이메일 형식입니다.",
  REQUIRED_PASSWORD: "비밀번호를 입력해주세요.",
  WEEK_PASSWORD: "비밀번호를 8자 이상 입력해주세요.",
  MISMATCHED_PASSWORD: "비밀번호가 일치하지 않습니다.",
  REQUIRED_NICKNAME: "닉네임을 입력해주세요.",
};

const checkPasswordValidation = ({ value }) => {
  if(!value) return { 
    isError: false,
    message: VALIDATION_MESSAGE.REQURIED_EMAIL
   };
  
  if(password.length < 8) return { 
    isError: false,
    message: VALIDATION_MESSAGE.WEEK_PASSWORD
  };

  return { isError: false }
}

export const validatePassword = (inputElement, messageElement) => {
  const value = inputElement.value;
  const { isError, message } = checkPasswordValidation(value);
  
  if(isError) return  showMessage(
    message,
    messageElement,
    inputElement
  );

  hideMessage(messageElement, inputElement);
};
```